src/com/thunder_cut/graphics/feature/Brush.java

```diff
@@ -24,99 +26,38 @@ public void pressed(int xPos, int yPos, CanvasPixelInfo canvasPixelInfo, Color c
     public void dragged(int xPos, int yPos, CanvasPixelInfo canvasPixelInfo, Color color) {
         boolean isPlusX = (prevXPos < xPos);
         boolean isPlusY = (prevYPos < yPos);
-        int currentX = prevXPos;
-        int currentY = prevYPos;
+        currentX = prevXPos;
+        currentY = prevYPos;
         int deltaX = Math.abs(xPos - prevXPos);
         int deltaY = Math.abs(yPos - prevYPos);
+        boolean isMaxDeltaX = deltaX > deltaY;
 
-        if(deltaX == 0) {
-            while(currentY != yPos) {
+        if((deltaX == 0) || (deltaY == 0)) {
+            while((currentX != xPos) || (currentY != yPos)) {
                 canvasPixelInfo.setPixel(canvasPixelInfo.getWidth() * currentY + currentX, color);
 
-                if(isPlusY) {
-                    currentY++;
-                }
-                else {
-                    currentY--;
-                }
+                controlPosition(isMaxDeltaX, isMaxDeltaX ? isPlusX : isPlusY);
             }
 
             return;
         }
 
-        if(deltaY == 0) {
-            while(currentX != xPos) {
-                canvasPixelInfo.setPixel(canvasPixelInfo.getWidth() * currentY + currentX, color);
-
-                if(isPlusX) {
-                    currentX++;
-                }
-                else {
-                    currentX--;
-                }
-            }
-
-            return;
-        }
-
-        double ratio;
+        double ratio = Math.max(deltaX, deltaY) / Math.min(deltaX, deltaY);
```

Comment:
Math.max and min returns integer value.    
If you need "double" type, you should cast divisor or dividend to "double".